lib/webview.dart

```diff
@@ -25,38 +26,188 @@ class _ScreenerAppState extends State<ScreenerApp> {
   late String googleLoginUrl = "'$_screenerHomeUrl/auth/flutter/'";
   late String postParam = "{}";
   late String requestMethod = "'post'";
-  late DragGesturePullToRefresh dragGesturePullToRefresh;
   final GoogleSignIn _googleSignIn = GoogleSignIn();
   late bool googleUser;
+  bool _lightMode = true;
+  bool firstLoad = true;
+
   @override
   void initState() {
     super.initState();
-    dragGesturePullToRefresh = DragGesturePullToRefresh();
+    //RazorPay settings
     _razorpay.on(Razorpay.EVENT_PAYMENT_SUCCESS, _handlePaymentSuccess);
     _razorpay.on(Razorpay.EVENT_PAYMENT_ERROR, _handlePaymentError);
     _razorpay.on(Razorpay.EVENT_EXTERNAL_WALLET, _handleExternalWallet);
+
+    final WebViewController controller =
+        WebViewController.fromPlatformCreationParams(
+            const PlatformWebViewControllerCreationParams());
+    // #enddocregion platform_features
+    controller
+      ..setJavaScriptMode(JavaScriptMode.unrestricted)
+      ..setNavigationDelegate(
+        NavigationDelegate(
+          onWebResourceError: (WebResourceError error) {
+            debugPrint('''
+                Page resource error:
+                code: ${error.errorCode}
+                description: ${error.description}
+                errorType: ${error.errorType}
+                isForMainFrame: ${error.isForMainFrame}
+          ''');
+          },
+          onPageStarted: (String url) async {
+            if (url.contains('screener')) {
+              if (firstLoad == true) {
+                final cookieString = await controller
+                    .runJavaScriptReturningResult('document.cookie');
+                final cookies = cookieString.toString().split(';');
+                for (final cookie in cookies) {
+                  final parts = cookie.split('=');
+                  final name = parts[0].trim();
+                  final value = parts.length > 1 ? parts[1].trim() : '';
+
+                  if (name == 'theme') {
+                    if (value.trim().contains("dark")) {
+                      setState(() {
+                        _lightMode = false;
+                      });
+                    }
+                    break;
+                  }
+                }
+                setState(() {
+                  firstLoad = false;
+                });
+              }
+              await controller.runJavaScript(
+                  "if(document.querySelectorAll('.top-nav-holder')){"
+                  "const topNavHolders = document.querySelectorAll('.top-nav-holder');"
+                  "for (let i = 0; i < topNavHolders.length; i++) {"
+                  "  topNavHolders[i].style.display = 'none';}}");
```

Comment:
For a second two headers are seen before this hides one. One in the app bar and one of the top of the page.

Also, this repeated section can be moved to a common function if needed.